src/redis_ai_objects/script.c

```diff
@@ -197,30 +120,59 @@ int RedisAI_ScriptRun_IsKeysPositionRequest_ReportKeys(RedisModuleCtx *ctx,
     return REDISMODULE_OK;
 }
 
-RedisModuleType *RAI_ScriptRedisType(void) { return RedisAI_ScriptType; }
-
-int RAI_ScriptRunAsync(RAI_ScriptRunCtx *sctx, RAI_OnFinishCB ScriptAsyncFinish,
-                       void *private_data) {
-
-    RedisAI_RunInfo *rinfo = NULL;
-    RAI_InitRunInfo(&rinfo);
-
-    rinfo->single_op_dag = 1;
-    rinfo->OnFinish = (RedisAI_OnFinishCB)ScriptAsyncFinish;
-    rinfo->private_data = private_data;
-
-    RAI_DagOp *op;
-    RAI_InitDagOp(&op);
-
-    op->commandType = REDISAI_DAG_CMD_SCRIPTRUN;
-    op->devicestr = sctx->script->devicestr;
-    op->sctx = sctx;
-
-    rinfo->dagOps = array_append(rinfo->dagOps, op);
-    rinfo->dagOpCount = 1;
-    if (DAG_InsertDAGToQueue(rinfo) != REDISMODULE_OK) {
-        RAI_FreeRunInfo(rinfo);
+int RedisAI_ScriptExecute_IsKeysPositionRequest_ReportKeys(RedisModuleCtx *ctx,
+                                                           RedisModuleString **argv, int argc) {
+    RedisModule_KeyAtPos(ctx, 1);
+    size_t argpos = 3;
+    if (argpos >= argc) {
         return REDISMODULE_ERR;
     }
-    return REDISMODULE_OK;
+    long long count;
+    while (argpos < argc) {
+        const char *str = RedisModule_StringPtrLen(argv[argpos++], NULL);
+
+        // Inputs, outpus, keys, lists.
+        if ((!strcasecmp(str, "INPUTS")) || (!strcasecmp(str, "OUTPUTS")) ||
```

Comment:
keys might be missing
But this is the parsing function responsibility. This is not the purpose of this function. Here it is a quick validation of the command structure with respect to keyspace. It should be aware that it can get anything from `{place_holder} n str1...strn` where `place_holder` is `INPUTS, KEYS, OUTPUTS, LIST_INPUTS`